### PR TITLE
Samtools improvements

### PR DESCRIPTION
**Pull request by robinandeer**
Add a bargraph plot for mapped/unmapped reads similar to that of "star"-module

See #251

### CODE REVIEW

**Comment by robinandeer** [comment]
- [ ] Use mixins for Picard and

File: multiqc/modules/samtools/stats.py [hunk 1]
```diff
@@ -10,125 +10,158 @@
...
-                        parsed_data['{}_percent'.format(k)] = (parsed_data[k] / parsed_data['raw_total_sequences']) * 100
+class ParseReportMixin():
+
+    def parse_reports(self):
+        """ Find Samtools stats logs and parse their data """
+
+        self.samtools_stats = dict()
+        for f in self.find_log_files(config.sp['samtools']['stats']):
+            parsed_data = dict()
+            relevant_lines = [line for line in f['f'].splitlines()
```
Comment (ewels) [inline multiqc/modules/samtools/stats.py]: Any reason why doing it this way is better?

Comment (robinandeer) [inline multiqc/modules/samtools/stats.py]: oh not really but I find it easier to read when one block does one thing :)

**Comment by robinandeer** [comment]
I've skipped a couple modules for now bc they have a lot more stuff in them that I'm sure if this is the right approach for them :P

**Comment by ewels** [comment]
This has suddenly become really huge quite quickly. Probably a bit late now, but any chance that you could remove the mixin stuff from this PR so that I can see what else has changed? Can do the mixin thing in a separate PR afterwards..

**Comment by robinandeer** [comment]
yeah sry about that - noticed it grew quite quickly - I'll separate it out somehow
maybe open a couple new PRs :P

**Comment by ewels** [comment]
Awesome, thanks 👍

**Pull request closed without merging**